src/localization/translationProvider.ts

```diff
@@ -0,0 +1,18 @@
+let translation: any = null;
+
+export const L = (key: string): string => {
+  if (!translation) {
+    translation = getTranslation();
+  }
+  
+  return translation[key] ?? key;
+};
+
+const getTranslation = (): any => {
+  try {
+    const browserLanguage = navigator.language.split('-')[0];
+    return require(`./${browserLanguage}`).translations;
```

Comment:
Dynamically using require is an anti-pattern and is forbidden by many linter configurations, we're not going to be able to keep this for our production code. There is a very similar approach which you can use, React recommends it, as a matter of fact, for when you want to lazy load components.

```js
const { translations } = await import(browserLanguage);
```

This `import` function is meant to be used in cases such as this one, `require` is an equivalent of the `import` keyword and should therefore only be used at the beginning of a file and should only ever receive constant strings, never variables. Just in case you're interested, [here](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-dynamic-require.md)'s the eslint rule that enforces this.